cfe_internal/enterprise/federation/federation.cf

```diff
@@ -187,12 +232,19 @@ bundle agent federation_manage_files
 bundle agent postgres_config
 # @brief Customize postgres config for superhub
 {
+  vars:
+    am_superhub::
+      "c[shared_buffers]" string => "1GB";
+      "c[max_locks_per_transaction]" string => "4000";
+      
   files:
     am_superhub::
       "/var/cfengine/state/pg/data/postgresql.conf"
-        edit_line => default:insert_lines("shared_buffers=1GB");
-      "/var/cfengine/state/pg/data/postgresql.conf"
-        edit_line => default:insert_lines("max_locks_per_transaction=4000");
+        edit_line => default:set_line_based( "$(this.bundle).c",
+                                     "=",
+                                     "\s*=\s*",
+                                     ".*",
+                                     "");
```

Comment:
@nickanderson it might be better if I don't have this postgres change in here since you have it in your PR: https://github.com/cfengine/masterfiles/pull/1369

Reply:
It doesn't matter, i will rebase